p2p/src/sync/tests/mod.rs

```diff
@@ -61,11 +63,17 @@ where
     );
     tokio::spawn(async move { man.main().await });
 
-    let config = Arc::new(common::chain::config::create_unit_test_config());
-    let (conn, sync) = T::start(addr, Arc::clone(&config), Default::default()).await.unwrap();
+    let chain_config = Arc::new(common::chain::config::create_unit_test_config());
+    let p2p_config = Arc::new(P2pConfig {
+        request_timeout: 1.into(),
+        ..Default::default()
```

Comment:
Please don't use the pattern `..Default..default()`... it's even preferable that you get rid of any instances of that you see around.

Reply:
I understand your concerns and I don't mind removing `..Default..default()` from all p2p code (all of them are in the tests code by the way), but what if we decide to change the default values in the config? In that case all tests will use some hardcoded values that are no longer default. Is that OK?

Reply:
Theoretically, the values should be chosen per object and per test. So, if the test is meant to use the default configuration, then that's OK, but if not, the full settings that get the test to function should be selected. There's no way to win all possible scenarios. I'd rather manually choose new values when adding a new member that an unrealistic blanket untraceable changes on tests just because the default was changed.

Reply:
Makes sense. I have removed all `..Default::default` usages from the p2p code.